### PR TITLE
Feature/generic types

### PR DESCRIPTION
**Pull request by UPovilas**
Added implementation of generic types: added two generic constraints;
Changes: the leaderboard now is able to interactively show additional information about the player: when the player's username is clicked, a pop-up window shows with additional information about the average player's WPM and accuracy. This is all done for the proper use of generic class StatisticsAnalyzer.

### CODE REVIEW

File: Typeracer/Services/StatisticsAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace Typeracer.Services;
+
+public class StatisticsAnalyzer<T> where T: class, IComparable<T>
```
Comment (elvinukas) [inline Typeracer/Services/StatisticsAnalyzer.cs:3]: yeah, this is okay, but what other class other than StatisticsModel could you analyze using this class in our program? it seems very one sided

**Review by RytasV — DISMISSED** [review]
Good job!

**Review by RytasV — DISMISSED** [review]
Approving PR

**Review by elvinukas — APPROVED** [review]

**Review by elvinukas — DISMISSED** [review]

**Review by elvinukas — DISMISSED** [review]

**Review by RytasV — APPROVED** [review]
Approving once again

**Review by elvinukas — APPROVED** [review]

**Pull request merged**